static/js/message_list.js

```diff
@@ -339,7 +344,7 @@ export class MessageList {
         this.view.clear_rendering_state(false);
         this.view.update_render_window(this.selected_idx(), false);
 
-        if (this === narrowed) {
+        if (this === message_lists.current && message_lists.is_current_narrowed()) {
```

Comment:
So this class of check previously combined two things:
* First, is this a narrowed message list at all? We can do that with `this.is_narrowed`, set in the constructor.
* Second, is this the current view? I think we can add a new function like this, and use it for the call points that want to know that.

```
is_current_view() {
   // Because Zulip has the invariant that at most two MessageList objects exist at any time,
   if (recent_topics_util.is_visible()) {
        return false;
   }
   if (narrow_state.is_active()) {
      return this.narrowed;
   } else {
      return !this.narrowed;
   }
}
```

Then these checks can be, for example, `this.table_name === "zhome" && this.is_current_view()`.

And then the 3 references you have in this file would use a either just `is_current_view` or `this.is_current_view() && this.is_narrowed`, as appropriate.

Reply:
Updated based on this comment!